dapps/v2Explorer/src/utils/ExplorerUtils.tsx

```diff
@@ -0,0 +1,107 @@
+import {Alert, Linking} from 'react-native';
+
+// @ts-expect-error - `@env` is a virtualised module via Babel config.
+import {ENV_PROJECT_ID} from '@env';
+
+function formatNativeUrl(appUrl: string, wcUri: string): string {
+  let safeAppUrl = appUrl;
+  if (!safeAppUrl.includes('://')) {
+    safeAppUrl = appUrl.replaceAll('/', '').replaceAll(':', '');
+    safeAppUrl = `${safeAppUrl}://`;
+  }
+  const encodedWcUrl = encodeURIComponent(wcUri);
+  return `${safeAppUrl}wc?uri=${encodedWcUrl}`;
+}
+
+function formatUniversalUrl(appUrl: string, wcUri: string): string {
+  let plainAppUrl = appUrl;
+  if (appUrl.endsWith('/')) {
+    plainAppUrl = appUrl.slice(0, -1);
+  }
+  const encodedWcUrl = encodeURIComponent(wcUri);
+
+  return `${plainAppUrl}/wc?uri=${encodedWcUrl}`;
+}
+
+export const navigateDeepLink = async (
+  universalLink: string,
+  deepLink: string,
+  wcURI: string,
+) => {
+  let tempDeepLink;
+
+  if (universalLink && universalLink !== '') {
+    tempDeepLink = formatUniversalUrl(universalLink, wcURI);
+  } else {
+    tempDeepLink = formatNativeUrl(deepLink, wcURI);
+  }
+
+  // if (
+  //   appLink &&
+  //   appLink === '' &&
+  //   !appLink.includes('https://') &&
+  //   !appLink.includes('http://')
+  // ) {
+  //   tempDeepLink = formatNativeUrl(appLink, wcURI);
+  // } else {
+  //   tempDeepLink = formatUniversalUrl(appLink, wcURI);
+  // }
+
+  try {
+    // Note: Could not use .canOpenURL() to check if the app is installed
+    // Due to having to add it to the iOS info
+    await Linking.openURL(tempDeepLink);
+  } catch (error) {
+    Alert.alert(`Unable to open this DeepLink: ${tempDeepLink}`);
+  }
+};
+
+export const fetchInitialWallets = async (
```

Comment:
Don't think `async` keyword is needed if you're not using `await` inside (since you're using `.then/.catch`)